### PR TITLE
docs: clarify the usage of the server information endpoint

### PR DESCRIPTION
**Pull request by rodrigc**
Discussed in [case 01501264](https://support.elastic.co/cases/5008X00002UnA9UQAV)

Signed-off-by: Craig Rodrigues <rodrigc@crodrigues.org>

## Motivation/summary

I found the existing documentation for the server information endpoint to be a bit confusing.
I didn't understand that GET and POST behave differently.

## Checklist

- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)
- [X] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

### CODE REVIEW

File: docs/api-info.asciidoc [hunk 1]
```diff
@@ -8,22 +8,55 @@ This lightweight endpoint is useful as a server up/down health check.
...
 ------------------------------------------------------------
 
-This endpoint always returns an HTTP 200.
+Sending an `HTTP GET` request to the server information endpoint
+will return an HTTP 200, indicating that the server is up,
+with an empty response body.
+
+Sending an `HTTP POST` request to the server information endpoint
+will return an HTTP 200, indicating that the server is up,
+with the body containing basic server information.
```
Comment (axw) [inline docs/api-info.asciidoc:24]: It's not GET vs. POST that makes the difference, it's whether the request has a valid auth token.

**Review by axw — COMMENTED** [review]
Thanks for sending the PR @rodrigc! I can't see your support case, but I guess there was some miscommunication: the endpoint's response body shouldn't differ based on whether it's a GET or POST method, what matters is whether the client is authorized. This is to minimise leaking information about the service to unauthorized (and potentially malicious) clients.

File: docs/api-info.asciidoc (hunk 1 above)
Comment (rodrigc) [reply docs/api-info.asciidoc:24]: Oh OK, the support agent communicated that GET would only return an HTTP 200 response,
so looks like the support agent was confused by these docs also.

In my support ticket, I communicated that I am using Elastic APM 8.10.3, and that I sent an authenticated request using

```
curl --verbose --request POST --header 'Authorization: ApiKey XXX' http://localhot:8200
```

and that just returned an HTTP 200 response, and did not return a payload.

I guess that led to confusion on both my part and the part of the support agent in [case 01501264](https://support.elastic.co/cases/5008X00002UnA9UQAV)

Comment (rodrigc) [reply docs/api-info.asciidoc]: @axw OK, I revised this patch, and clarified the text around the examples

Comment (axw) [reply docs/api-info.asciidoc]: Looks great, thank you :)

**Review by axw — APPROVED** [review]

**Comment by axw** [comment]
@rodrigc sorry, could you please sign your commit and force push again?

**Comment by rodrigc** [comment]
Done

**Review by axw — APPROVED** [review]

**Comment by rodrigc** [comment]
@axw can you trigger the docs CI?  I'm not sure, but I think something like:

@elasticsearchmachine run elasticsearch-ci/docs

is required

**Comment by rodrigc** [comment]
@axw Thanks for the review and merge of this PR.

I have one question.  Would it be possible to include a default role in the apm-server code which is similar to
the one mentioned here:
https://www.elastic.co/guide/en/apm/guide/current/api-key.html#create-api-key-user

That way, the end user would not need to worry so much about creating a role
to use an API key which works with APM

**Comment by axw** [comment]
@rodrigc we could predefine a role in Elasticsearch (not in apm-server), but you would still need to create/assign a user that role, so it doesn't seem entirely ideal either.

I think the long-term vision is that this should be built into a standard, but broader, role like "editor". The idea being that you have users that are admins, viewers, or editors; admins and editors should be able to create API Keys for agents, whereas viewers should not. If you wanted a more fine-grained role, then you would still have the option to create one.

Does that sound sensible to you?

**Comment by rodrigc** [comment]
That seems like a reasonable approach.  I agree that more thought needs to be put into this.

**Comment by axw** [comment]
@rodrigc I've opened https://github.com/elastic/apm/issues/831 to track this

**Pull request merged**